bezier-rs/lib/src/lib.rs

```diff
@@ -331,6 +331,67 @@ impl Bezier {
 		};
 		bezier_starting_at_t1.split(adjusted_t2)[t2_split_side]
 	}
+
+	/// Find the roots of the linear equation ax + b
+	fn solve_linear(a: f64, b: f64) -> Vec<f64> {
+		let mut roots = Vec::new();
+		if b != 0. {
+			roots.push(-b / a);
+		}
+		roots
+	}
+
+	/// Find the roots of the linear equation ax^2 + bx + c
+	fn solve_quadratic(discriminant: f64, two_times_a: f64, b: f64, c: f64) -> Vec<f64> {
```

Comment:
Can you include a note saying that `discriminant` and `two_times_a` are passed in as parameters so that the computation can be done before hand using the `DVec2`s?